timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -308,6 +298,106 @@ void declare_context(py::module &m) {
             Coordinates have shape (F, N, 3)
             Boxes have shape (F, 3, 3)
 
+        Note: All boxes returned will be identical as local md only runs under NVT.
+    )pbdoc")
+        .def(
+            "multiple_steps_local_selection",
+            [](timemachine::Context &ctxt,
+               const int n_steps,
+               const int reference_idx,
+               const py::array_t<int, py::array::c_style> &selection_idxs,
+               const int burn_in,
+               const int store_x_interval,
+               const double radius,
+               const double k) -> py::tuple {
+                if (n_steps <= 0) {
+                    throw std::runtime_error("local steps must be at least one");
+                }
+                if (burn_in < 0) {
+                    throw std::runtime_error("burn in steps must be greater or equal to zero");
+                }
+                timemachine::verify_local_md_parameters(radius, k);
+
+                const int N = ctxt.num_atoms();
+                const int x_interval = (store_x_interval <= 0) ? n_steps : store_x_interval;
+
+                if (reference_idx < 0 || reference_idx >= N) {
+                    throw std::runtime_error("reference idx must be at least 0 and less than " + std::to_string(N));
+                }
+                std::vector<int> vec_selection_idxs(selection_idxs.size());
+                std::memcpy(vec_selection_idxs.data(), selection_idxs.data(), vec_selection_idxs.size() * sizeof(int));
+                verify_atom_idxs(N, vec_selection_idxs);
+                std::set<int> selection_set(vec_selection_idxs.begin(), vec_selection_idxs.end());
+                if (selection_set.find(reference_idx) != selection_set.end()) {
+                    throw std::runtime_error("reference idx must not be in selection idxs");
+                }
+
+                std::array<std::vector<double>, 2> result = ctxt.multiple_steps_local_selection(
+                    n_steps, reference_idx, vec_selection_idxs, burn_in, x_interval, radius, k);
+                const int D = 3;
+                const int F = result[0].size() / (N * D);
```

Comment:
nit: could this be `n_free_particles` or similar?

Reply:
If I understand correctly, you are asking if we could only return the coordinates that actually change? If so we could, not sure the performance would be worth it, but would have to investigate more thoroughly.

Reply:
No, sorry, I was just suggesting a more descriptive name instead of `F` (but maybe I misunderstood what `F` represents?)

Reply:
Ah, I see, `F` is frames and we use it elsewhere. No action needed.

Reply:
F = # number of frames, but can switch it to `num_frames`